docs/content/tutorial/saving-your-data.mdx

```diff
@@ -0,0 +1,184 @@
+  likely you have more of. In-memory processing is faster, but it can only be
+  done on a computer's RAM and you likely have less RAM on your machines.
+</Note>
+
+---
+
+## Step 1: Writing files to storage
+
+Now, you’ll write your files to a more permanent location. For this tutorial, you’ll be writing to another directory in your file system by using the `fs_io_manager`, which is bundled with the core `dagster` package.
+
+In `__init__.py`, add the following snippet anywhere above the `def = Definitions(...)` line:
+
+```python
+from dagster import (
+    AssetSelection,
+    Definitions,
+    ScheduleDefinition,
+    define_asset_job,
+    load_assets_from_modules,
+    fs_io_manager # Update the imports at the top of the file to also include this
+)
+
+io_manager = fs_io_manager.configured({
+    "base_dir": "data", # Path is built relative to where `dagster dev` is run
+})
+```
+
+Next, update your `Definitions` (shown below) to attach the I/O manager to your code location by defining it as a **resource**.
+
+```python
+defs = Definitions(
+    assets=hackernews_assets,
+    schedules=[hackernews_schedule],
+    resources={
+        "io_manager": io_manager,
+    }
+)
+```
+
+<Note>
+  <strong>About resources</strong>
+  <br />
+  I/O managers are a special type of resource, a Dagster concept for{" "}
+  <strong>connecting</strong> to external services. I/O managers are resources
+  made specifically to control reading and writing to external services. Refer
+  to the <a href="/concepts/resources">Resources documentation</a> for more
+  information.
+</Note>
+
+To see the effect of this change, reload your code location in the Dagster UI and materialize your assets. When you click on an asset, you’ll see the `path` metadata pointing to the `data` directory on your computer.
+
+Look at your Dagster project in your file system (via terminal, Finder, or Windows Explorer) and you’ll find a new directory called `data` that contains your materialized assets.
+
+This is a short example of the value of I/O managers. In a few lines of code, assets previously stored in ephemeral memory can now be saved to many locations. With I/O managers, you can save your assets to places like AWS S3, Google Cloud Storage, or Azure Blob Storage.
+
+In addition to file storage systems, I/O managers can connect to any service from which data can be read or written, such as databases.
+
+---
+
+## Step 2: Writing to databases
+
+In data engineering, a common asset is tabular data. These are often seen as database tables or DataFrames in-memory. In Dagster, I/O managers define the relationship between how data is saved as a table in the database and what in-memory data format to represent the data as.
+
+Dagster includes out-of-the-box IO managers for common databases, such as DuckDB. When a DataFrame is returned in an asset’s definition, it is translated into a database-compatible format. The I/O manager then runs a SQL query to write the data into the database.
```

Comment:
```suggestion
Dagster includes out-of-the-box I/O managers for common databases, such as DuckDB. When a DataFrame is returned in an asset’s definition, it is translated into a database-compatible format. The I/O manager then runs a SQL query to write the data into the database.
```

I'm annoyed with myself for not catching this before now 🤦‍♀️

Reply:
How's the Grammarly configuring goin'? You shouldn't have to be annoyed at yourself, but even then, we should have Grammarly catch this.